### PR TITLE
7-ticket is done with error handling and TDD

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
                 });
         });
     });
+    describe("POST /api/articles/:article_id/comments", () => {
+        test("200: should return 200 and insert/post to add new record to the comments table", () => {
```
Comment (yalicee) [inline __tests__/app.test.js:159]: Should be another 201 that ignores unnecessary properties

File: __tests__/app.test.js [hunk 2]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                    body: {
+                        article_id: 9,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { comments } = body;
```
Comment (yalicee) [inline __tests__/app.test.js:174]: should be on key of  comment (singular)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                    expect(comments).toHaveProperty("comment_id", expect.any(Number));
+                    expect(comments).toHaveProperty("body", expect.any(String));
+                    expect(comments).toHaveProperty("article_id", expect.any(Number));
+                    expect(comments).toHaveProperty("author", expect.any(String));
+                    expect(comments).toHaveProperty("votes", expect.any(Number));
+                    expect(comments).toHaveProperty("created_at", expect.any(String));
```
Comment (yalicee) [inline __tests__/app.test.js:180]: You have most of the actual comment data so test for that

File: __tests__/app.test.js [hunk 4]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
+        test("400 should be returned when the property article_id of the body cause foreign key relationship problem", () => {
```
Comment (yalicee) [inline __tests__/app.test.js:221]: not quite sure what this is testing for?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                .expect(200)
+                .send({
+                    user_name: "butter_bridge",
+                    body: {
+                        article_id: 9,
+                        author: "butter_bridge",
+                        votes: 20,
+                        created_at: new Date(),
+                        body: "post request will going to be ok",
+                    },
```
Comment (yalicee) [inline __tests__/app.test.js:171]: This comment body should have only 2 properties. username and body

File: __tests__/app.test.js [hunk 6]
```diff
@@ -155,4 +155,106 @@ describe("All requests", () => {
...
+                })
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Invalid request!");
+                });
+        });
+        test("400 should be returned when the property author of the body cause foreign key relationship problem", () => {
```
Comment (yalicee) [inline __tests__/app.test.js:240]: This test description can be changed too

File: models/check-exist-item.js [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+             WHERE ${column} = $1;`,
             [article_id]
         )
         .then(({ rows }) => {
             if (!rows.length) {
                 return Promise.reject({ status: 404, msg: "Not found!" });
             }
-            return rows;
+            // return rows;
         });
```
Comment (yalicee) [inline models/check-exist-item.js:15]: article_id is also a foreign key so again, a separate check that the article exists should not be necessary as a psql error will be thrown.

File: endpoints.json [hunk 8]
```diff
@@ -57,5 +57,20 @@
...
+        "description": "It will us to post/insert new comment to comments table by article_id and requested body",
+        "queries": [],
+        "exampleResponse": {
+            "comments": [
+                {
+                    "body": "Oh, I've got compassion running out of my nose, pal! I'm the Sultan of Sentiment!",
+                    "votes": 16,
+                    "author": "butter_bridge",
+                    "article_id": 9,
+                    "created_at": 1586179020000
```
Comment (yalicee) [inline endpoints.json:71]: I believe created_at data type will not be integers

File: controllers/articles.controller.js [hunk 9]
```diff
@@ -38,3 +40,24 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    const { user_name, body } = req.body;
+
+    const promises = [insertCommentsByArticleId(article_id, body)];
+
+    if (article_id) {
+        promises.push(checkArticleExist(article_id, "articles", "article_id"));
+    }
+    if (user_name) {
+        promises.push(checkArticleExist(user_name, "users", "username"));
+    }
```
Comment (yalicee) [inline controllers/articles.controller.js:55]: author in the comments table is a foreign key which relates to username in the users table. A psql error will be thrown if this is violated. It is not necessary to separately check if the user exists.

**Review by yalicee — CHANGES_REQUESTED** [review]
Please refactor with all the changes I have mentioned

**Comment by Hassen-Ahmed** [comment]
Thank you. I will fix it and ask pull request again!

**Pull request closed without merging**